sql/plan.go

```diff
@@ -50,6 +49,10 @@ func (p *planner) resetTxn() {
 // plan needs to be iterated over using planNode.Next() and planNode.Values()
 // in order to retrieve matching rows.
 func (p *planner) makePlan(stmt parser.Statement) (planNode, error) {
+	if stmt.StatementType() == parser.DDL {
+		p.txn.SetSystemDBTrigger()
```

Comment:
that should be fine in practice. It would generate false refreshes, but that should be rare enough. Testing might be a bit trickier if this still ends up in a committed transaction at the kv layer.
We also have non-DDL operations that needs to set the trigger (eg: insert checks if it's operating on a system table, and delete/update should do the same), but I'll cover that with  #2502